src/lib/core/PeerId.h

```diff
@@ -68,4 +69,22 @@ class PeerId
     CompressedFabricId mCompressedFabricId = kUndefinedCompressedFabricId;
 };
 
+class ScopedNodeId
```

Comment:
What's the broader picture here in terms of using this vs `PeerId`? In which circumstances would one be preferred over the other? Also, does the scoping work uniformly on both client and server side constructs? This needs to be well documented.

I'd presume that PeerId is mostly relegated to things to do with operational discovery, while ScopedNodeId would take over as the preferred construct everywhere else.

Lastly, will there be follow-on PRs to update existing instances of PeerId that _should_ get switched to ScopedNodeId?

Reply:
I also don't think `ScopedNodeId` should live in the same file as `PeerId`. I'd think `PeerId` should move to somewhere more within the confines of the DNS-SD resolution header files (and within core), while `ScopedNodeId` should go into a file of its own.

Reply:
I found 2 usage of compressed fabric id:

1. Announce mDNS name
2. Derive group encryption key

Generally, compressed fabric id have limited usage than FabricIndex. There will Following PRs move towards ScopedNodeId gradually.

I have moved ScopedNodeId into a separate file.

Reply:
Would love to get some more guidance here about the difference between the PeerId and ScopedNodeId. Exactly where do we use one vs. the other? Why do they need to be completely separate? Are the fabric index and compressed fabric id not related? Seems like you could instantiate with either and have the other information available easily.

Reply:
@mrjerryjohns @cecille Here is a PR depending on this one https://github.com/project-chip/connectedhomeip/pull/16250

`ScopedNodeId` is more suitable here, because compressed id is not easily accessible from session, and it is unique.

Reply:
@cecille A compressed fabric id (and hence PerrId) is not actually guaranteed to be unique, for one thing, thought it's globally-unlikely-to-collide.

A ScopedNodeId has no global uniqueness properties at all, but is unique within the context of a single node.  So it's a much better key for keying node-local state of various sorts.